packages/react/src/components/teleport/util.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Companion utility for <Teleport />
+ * 
+ * Searches the DOM for a teleport target element
+ * with id [teleportId]. If none is found, creates
+ * one at the root of the DOM.
+ * 
+ * @param teleportKey the key of teleport target
+ * 
+ * @returns DOM selector for teleport target i.e. #foo
+ */
+export const findOrCreateTeleportTarget = (teleportKey: string): HTMLElement => {
+  const element = globalThis.document.getElementById(teleportKey)
```

Comment:
Wouldn't `teleportElementId` or `teleportTargetId` be better since we use it here as an element ID?

Reply:
This feels like leaking an implementation detail, the idea of a "teleport key" is to abstract this detail away.

You choose an identifier and the rest if taken care for you

Reply:
Based [on this](https://github.com/slashid/javascript/pull/141#discussion_r1308514522) I think we're ok with it as it is.